skglm/estimators.py

```diff
@@ -1537,3 +1539,135 @@ def path(self, X, Y, alphas, coef_init=None, return_n_iter=False, **params):
             ws_strategy=self.ws_strategy, fit_intercept=self.fit_intercept,
             warm_start=self.warm_start, verbose=self.verbose)
         return solver.path(X, Y, datafit, penalty, alphas, coef_init, return_n_iter)
+
+
+class GroupLasso(LinearModel, RegressorMixin):
+    r"""GroupLasso estimator based on Celer solver and primal extrapolation.
+
+    The optimization objective for GroupLasso is:
+
+    .. math::
+        1 / (2 xx n_"samples") \sum_g ||y - X_{[g]} w_{[g]}||_2 ^ 2 + alpha \sum_g
+        weights_g ||w_{[g]}||_2
+
+    with :math:`w_{[g]}` (respectively :math:`X_{[g]}`) being the coefficients
+    (respectively the columns) of the g-th group.
+
+    Parameters
+    ----------
+    groups : int | list of ints | list of lists of ints
+        Partition of features used in the penalty on ``w``.
+        If an int is passed, groups are contiguous blocks of features, of size
+        ``groups``.
+        If a list of ints is passed, groups are assumed to be contiguous,
+        group number ``g`` being of size ``groups[g]``.
+        If a list of lists of ints is passed, ``groups[g]`` contains the
+        feature indices of the group number ``g``.
+
+    alpha : float, optional
+        Penalty strength.
+
+    weights : array, shape (n_groups,), optional (default=None)
+        Positive weights used in the L1 penalty part of the Lasso
+        objective. If ``None``, weights equal to 1 are used.
+
+    max_iter : int, optional (default=50)
+        The maximum number of iterations (subproblem definitions).
+
+    max_epochs : int, optional (default=50_000)
+        Maximum number of CD epochs on each subproblem.
+
+    p0 : int, optional (default=10)
+        First working set size.
+
+    verbose : bool or int, optional (default=0)
+        Amount of verbosity.
+
+    tol : float, optional (default=1e-4)
+        Stopping criterion for the optimization.
+
+    positive : bool, optional (defautl=False)
+        When set to ``True``, forces the coefficient vector to be positive.
+
+    fit_intercept : bool, optional (default=True)
+        Whether or not to fit an intercept.
+
+    warm_start : bool, optional (default=False)
+        When set to ``True``, reuse the solution of the previous call to fit as
+        initialization, otherwise, just erase the previous solution.
+
+    ws_strategy : str, optional (default="subdiff")
+        The score used to build the working set. Can be ``fixpoint`` or ``subdiff``.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,)
+        parameter vector (:math:`w` in the cost function formula)
+
+    sparse_coef_ : scipy.sparse matrix, shape (n_features, 1)
```

Comment:
We don't define this attribute in ``_glm_fit``, hence it not accessible to all estimators including the ``GroupLasso``

To retain compatibility with scikit-learn, we should address it asap (ideally in a separate in PR)